hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -287,18 +344,30 @@ private void handlePutEvent(OMDBUpdateEvent<String, Object> event,
 
   private void handleSizeRelatedTablePutEvent(
       OMDBUpdateEvent<String, Object> event, String countKey,
-      String unReplicatedSizeKey, String replicatedSizeKey,
+      String unReplicatedSizeKey, String replicatedSizeKey, String tableName,
       HashMap<String, Long> objectCountMap,
       HashMap<String, Long> unreplicatedSizeCountMap,
-      HashMap<String, Long> replicatedSizeCountMap) {
+      HashMap<String, Long> replicatedSizeCountMap) throws IOException {
+
+    boolean isFileTable = tableName.equals(OPEN_FILE_TABLE);
+    boolean isKeyTable = tableName.equals(OPEN_KEY_TABLE);
+    boolean isDeletedDirTable = tableName.equals(DELETED_DIR_TABLE);
+
     // Handle PUT for OpenKeyTable & OpenFileTable
     if (event.getValue() instanceof OmKeyInfo) {
       OmKeyInfo omKeyInfo = (OmKeyInfo) event.getValue();
       objectCountMap.computeIfPresent(countKey, (k, count) -> count + 1L);
-      unreplicatedSizeCountMap.computeIfPresent(unReplicatedSizeKey,
-          (k, size) -> size + omKeyInfo.getDataSize());
-      replicatedSizeCountMap.computeIfPresent(replicatedSizeKey,
-          (k, size) -> size + omKeyInfo.getReplicatedSize());
+      if (isFileTable || isKeyTable) {
```

Comment:
We can reduce if else complexity here, as we know that size of tables are meant based on files/keys which can be computed later at bucket or volume level as well or can be computed for file and key table as well , try to keep the code generic as much as possible.